packages/front-end/components/Settings/ApiKeysModal.tsx

```diff
@@ -117,16 +141,22 @@ const ApiKeysModal: FC<{
           label="Role"
           value={form.watch("type")}
           onChange={(v) => form.setValue("type", v)}
-          options={[
-            {
-              label: "Admin",
-              value: "admin",
-            },
-            {
-              label: "Read-only",
-              value: "readonly",
-            },
-          ]}
+          options={roleOptions.map((r) => ({
```

Comment:
This change improves the select dropdown options to include the role description like how we do in the invite user modal. It adds context about what each role can do.